modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.pc.analytics.core.kpi.utils.DASConfigurationUtils;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Rest Client to communicate the analytics configuration details to the WSO2 BPS from PC
+ */
+public class BPSConfigRestClient {
+    private static final Log log = LogFactory.getLog(BPSConfigRestClient.class);
+
+    /**
+     * Send post request to a BPS BPMN rest web service
+     *
+     * @param dasConfigDetails is the request message that need to be sent to the web service
+     * @param processName
+     * @param processVersion
+     * @return the result as a String
+     */
+    public static void post(String dasConfigDetails, String processName, String processVersion)
+            throws IOException, XMLStreamException, RuntimeException {
+
+        String bpsurl = DASConfigurationUtils.getBPSURL();
+        RegistryUtils.setTrustStoreSystemProperties();
+        HttpClient httpClient = new HttpClient();
+        String requestUrl = bpsurl + AnalyticsConfigConstants.BPS_PROCESS_VAR_PUBLISH_REST_PATH + processName + "_"
+                + processVersion;
+
+        PostMethod postRequest = new PostMethod(requestUrl);
+        postRequest.setRequestHeader("Authorization", DASConfigurationUtils.getAuthorizationHeader());
+        StringRequestEntity input = new StringRequestEntity(dasConfigDetails, "application/json", "UTF-8");
+        postRequest.setRequestEntity(input);
+
+        int returnCode = httpClient.executeMethod(postRequest);
+
+        InputStreamReader reader = new InputStreamReader((postRequest.getResponseBodyAsStream()));
+        BufferedReader br = new BufferedReader(reader);
+
+        String output = null;
+        StringBuilder totalOutput = new StringBuilder();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Output from Server .... \n");
+        }
+
+        while ((output = br.readLine()) != null) {
+            totalOutput.append(output);
+        }
+
+        String responseMsg = totalOutput.toString();
+
+        postRequest.releaseConnection();
+        if (br != null) {
+            try {
+                br.close();
+            } catch (IOException e) {
+                String errMsg = "BPS Config Rest client BufferedReader close exception.";
+                log.error(errMsg, e);
+            }
+        }
+
+        //deal with the response
+        if (returnCode == HttpStatus.SC_OK) {
+            log.info("BPS was acknowleged the Analytics Configuration details");
+        } else if (returnCode == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+            log.info(responseMsg);
+            throw new RuntimeException(responseMsg);
+        } else {
+            String errMsg =
```

Comment:
Can't we handle both error cases in the same way. Also have to use error level logging.